includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -702,16 +714,75 @@ public function stats_boxes() {
 	 * Overloads the parent method
 	 *
 	 * @since  1.2.0
-	 * @return void
 	 */
 	public function no_items() {
-		if ( ! $this->view || 'users' == $this->view ) {
-			$type = 'learners';
+		$highlight = false;
+
+		if ( 'lessons' === $this->type && ! $this->get_course_filter_value() ) {
+			$message   = __( 'View your Lessons data by first selecting a course.', 'sensei-lms' );
+			$highlight = true;
 		} else {
-			$type = $this->view;
+			if ( ! $this->type || 'users' === $this->type ) {
+				$type = __( 'students', 'sensei-lms' );
+			} else {
+				$type = $this->type;
+			}
+
+			// translators: Placeholders %1$s and %3$s are opening and closing <em> tags, %2$s is the view type.
+			$message = sprintf( __( '%1$sNo %2$s found%3$s', 'sensei-lms' ), '<em>', $type, '</em>' );
 		}
-		// translators: Placeholders %1$s and %3$s are opening and closing <em> tages, %2$s is the view type.
-		echo wp_kses_post( sprintf( __( '%1$sNo %2$s found%3$s', 'sensei-lms' ), '<em>', $type, '</em>' ) );
+
+		?>
+		<div class="sensei-analysis__no-items-message <?php echo $highlight ? 'sensei-analysis__no-items-message--highlighted' : ''; ?>">
```

Comment:
Just curious if we need this `highlighted` class? Could we use the `sensei-analysis__no-items-message` selector instead?

Reply:
I do need it because we have two different message views now. Here's the case:

The highlighted message is centered and with bigger font:

<img width="1295" alt="Screenshot on 2022-03-09 at 19-58-56" src="https://user-images.githubusercontent.com/1612178/157502923-2601e306-76e6-4acf-9b3e-579b5b09e149.png">

The standard message is very similar to the default WP no-items message:

<img width="1292" alt="Screenshot on 2022-03-09 at 19-59-13" src="https://user-images.githubusercontent.com/1612178/157503149-edac3163-e9c7-4bd2-810b-4bc01563cc7f.png">

Reply:
I don't see a pressing need to preserve the styling for the other reports. Seems to me we could simplify a bit and use the same styling for Students and Courses too when they don't have any data. WDYT?

Reply:
Agree. Just to confirm - we will be using the `highlighted` version only. Did I understand correctly?

Reply:
I've updated the styling in f78655e85d13d3f2925dee2bd7c5c588a5d9df58. Let me know if I'm not on the same page. 🙂

Reply:
Yup, same book, chapter and page. 😄